File: src/tests/java/vertx/tests/busmods/workqueue/OrderQueue.java
package vertx.tests.busmods.workqueue;

import org.vertx.java.busmods.workqueue.WorkQueue;
import org.vertx.java.newtests.TestUtils;

/**
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class OrderQueue extends WorkQueue {

  private TestUtils tu = new TestUtils();

  public OrderQueue() {
    super("orderQueue", 30000);
  }

  @Override
  public void start() {
    super.start();
    tu.appReady();
  }

  @Override
  public void stop() {
    super.stop();
    tu.appStopped();
  }
}

File: src/tests/javascript/busmods/workqueue/order_queue.js
load('vertx.js');
load('test_utils.js');

var tu = new TestUtils();

var queue = new vertx.WorkQueue("test.orderQueue", 30000);
queue.start();
tu.appReady();

function vertxStop() {
  queue.stop();
  tu.checkContext();
  tu.appStopped();
}
